www/js/lib/uiUtil.js

```diff
@@ -45,35 +45,79 @@ define(rqDef, function(settingsStore) {
      */
     function systemAlert(message, label, isConfirm, declineConfirmLabel, approveConfirmLabel, closeMessageLabel) {
-        declineConfirmLabel = declineConfirmLabel || "Cancel";
-        approveConfirmLabel = approveConfirmLabel || "Confirm";
-        closeMessageLabel = closeMessageLabel || "Okay";
-        label = label || (isConfirm ? "Confirmation" : "Message");
+        declineConfirmLabel = declineConfirmLabel || 'Cancel';
+        approveConfirmLabel = approveConfirmLabel || 'Confirm';
+        closeMessageLabel = closeMessageLabel || 'Okay';
+        label = label || (isConfirm ? 'Confirmation' : 'Message');
         return new Promise(function (resolve, reject) {
-            if (!message) reject("Missing body message");
+            if (!message) reject('Missing body message');
             // Set the text to the modal and its buttons
-            document.getElementById("approveConfirm").textContent = approveConfirmLabel;
-            document.getElementById("declineConfirm").textContent = declineConfirmLabel;
-            document.getElementById("closeMessage").textContent = closeMessageLabel;
-            document.getElementById("modalLabel").textContent = label;
+            document.getElementById('approveConfirm').textContent = approveConfirmLabel;
+            document.getElementById('declineConfirm').textContent = declineConfirmLabel;
+            document.getElementById('closeMessage').textContent = closeMessageLabel;
+            document.getElementById('modalLabel').textContent = label;
             // Using innerHTML to set the message to allow HTML formatting
-            document.getElementById("modalText").innerHTML = message;
+            document.getElementById('modalText').innerHTML = message;
             // Display buttons acc to the type of alert
-            document.getElementById("approveConfirm").style.display = isConfirm ? "inline" : "none";
-            document.getElementById("declineConfirm").style.display = isConfirm ? "inline" : "none";
-            document.getElementById("closeMessage").style.display = isConfirm ? "none" : "inline";
+            document.getElementById('approveConfirm').style.display = isConfirm ? 'inline' : 'none';
+            document.getElementById('declineConfirm').style.display = isConfirm ? 'inline' : 'none';
+            document.getElementById('closeMessage').style.display = isConfirm ? 'none' : 'inline';
             // Display the modal
-            $("#alertModal").modal("show");
+            const modal = document.querySelector('#alertModal');
+            const backdrop = document.createElement('div');
+            backdrop.classList.add('modal-backdrop');
+            document.body.appendChild(backdrop);
+            // Set focus to the first focusable element inside the modal
+            modal.focus();
```

Comment:
@aritroCoder If you move this focus statement to the end of this function, then the modal will be properly focused. The problem is that it comes too early in the code. Then you can add key event listeners to the modal, rather than to document.

Reply:
done